wolfssl/wolfcrypt/port/Espressif/esp32-crypt.h

```diff
@@ -694,20 +769,25 @@ extern "C"
     WOLFSSL_LOCAL int esp_sha_hw_unlock(WC_ESP32SHA* ctx);
 
     /* esp_sha_hw_islocked: returns 0 if not locked, otherwise owner address */
-    WOLFSSL_LOCAL int esp_sha_hw_islocked(WC_ESP32SHA* ctx);
+    WOLFSSL_LOCAL uintptr_t esp_sha_hw_islocked(WC_ESP32SHA* ctx);
```

Comment:
Consider fixing `esp_sha_release_unfinished_lock`. It still expects `int` and then casts back to `uintptr_t`.

Reply:
Good catch. Yes, that should have been a `uintptr_t` all along.